src/gitops/utils/components/DetailsPageTitle/ResourceDetailsTitle.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom-v5-compat';
+import DevPreviewBadge from '../../../../components/import/badges/DevPreviewBadge';
+import { DEFAULT_NAMESPACE } from '../../../utils/constants';
+import { isApplicationRefreshing } from '../../../utils/gitops';
+import { useGitOpsTranslation } from '../../../utils/hooks/useGitOpsTranslation';
+import { Action, K8sModel, K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import { Breadcrumb, BreadcrumbItem, Spinner, Title } from '@patternfly/react-core';
+import ActionsDropdown from '../../../utils/components/ActionDropDown/ActionDropDown';
+import DetailsPageTitle, { PaneHeading } from './DetailsPageTitle';
+import FavoriteButton from '../../../components/appset/FavoriteButton';
+
+type ResourceDetailsTitleProps = {
+  obj: K8sResourceCommon;
+  model: K8sModel;
+  name: string;
+  namespace: string;
+  actions: Action[];
+  // Configurable properties for different resource types
+  iconText: string;
+  iconTitle: string;
+  resourcePrefix?: string; // e.g., "Argo CD" for Applications/ApplicationSets
+  showDevPreviewBadge?: boolean;
+  showRefreshSpinner?: boolean;
+};
+
+const ResourceDetailsTitle: React.FC<ResourceDetailsTitleProps> = ({
+  obj,
+  model,
+  name,
+  namespace,
+  actions,
+  iconText,
+  iconTitle,
+  resourcePrefix = '',
+  showDevPreviewBadge = true,
+  showRefreshSpinner = true,
+}) => {
+  const { t } = useGitOpsTranslation();
+  
+  return (
+    <>
+      <div>
+        <DetailsPageTitle
+          breadcrumb={
+            <Breadcrumb className="pf-c-breadcrumb co-breadcrumb">
+              <BreadcrumbItem>
+                <Link
+                  to={`/k8s/ns/${namespace || DEFAULT_NAMESPACE}/${
+                    model.apiGroup + '~' + model.apiVersion + '~' + model.kind
+                  }`}
+                >
+                  {resourcePrefix} {t(model.labelPlural)}
+                </Link>
+              </BreadcrumbItem>
+              <BreadcrumbItem>{resourcePrefix} {t(model.labelPlural + ' Details')}</BreadcrumbItem>
+            </Breadcrumb>
+          }
+        >
+          <PaneHeading>
+            <Title headingLevel="h1">
+              <span
+                className="co-m-resource-icon co-m-resource-icon--lg"
```

Comment:
I think the application-details-title.scss (in src/gitops/components/application) needs to be moved over to this folder and renamed resource-details-title.scss.  The original class name I had was argocd-application-icon, so call it argocd-resource-icon and add it to this here.  The text (A or AS) doesn't have the proper background colour.

I originally only had:

```
.argocd-application-icon {
    background-color: #E9654B;
}
```
Do we need all those styles that are currently in application-detials-title.scss?